app/javascript/context/subreducers/ApplicationDetailsSubreducer.js

```diff
@@ -0,0 +1,157 @@
+import { cloneDeep, findIndex } from 'lodash'
+
+import LEASE_STATES from 'context/actionCreators/application_details/leaseSectionStates'
+
+import ACTIONS from '../actions'
+
+export const getEmptyStatusModalState = () => ({
+  alertMsg: null,
+  header: null,
+  isOpen: false,
+  loading: false,
+  showAlert: false,
+  status: null,
+  substatus: null,
+  submitButton: null,
+  isInAddCommentMode: false
+})
+
+export const getEmptyApplicationDetailsState = () => ({
+  shortform: {
+    application: null,
+    fileBaseUrl: null
+  },
+  supplemental: {
```

Comment:
Empty state should enumerate all of the fields that will get populated, this just makes it a little easier to see what the shape of the state object looks like

I think navigating reducers will be way clearer in typescript since the shape of objects (and in action payloads!) is defined in the type

Reply:
this is a great change